### PR TITLE
fix #10093, rel to #10075: store mapthemes internal and fix npe on theme select

### LINKED ISSUE #10093: Crash after renaming theme file
While testing theme sync yesterday I had several crashes with 2021.02.28-NB.

Sadly I cannot really reproduce it up to now.
What I potentially did before this happened was:
- Debug mode active
- End c:geo
- Rename some theme folder and/or theme file
- Start c:geo
- Open live map
- Open theme selection menu

Not sure whether it crashed on startup, on live map or only when opening the theme selection.

This is the log I exported yesterday:
```
02-28 23:18:51.730 17335 17335 E AndroidRuntime: FATAL EXCEPTION: main
02-28 23:18:51.730 17335 17335 E AndroidRuntime: Process: cgeo.geocaching, PID: 17335
02-28 23:18:51.730 17335 17335 E AndroidRuntime: java.lang.NullPointerException: Attempt to read from field 'android.net.Uri cgeo.geocaching.storage.ContentStorage$FileInformation.uri' on a null object reference
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.v6.RenderThemeHelper.getThemeFor(RenderThemeHelper.java:183)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.v6.RenderThemeHelper.reapplyMapTheme(RenderThemeHelper.java:134)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.v6.NewMap.switchTileLayer(NewMap.java:642)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.v6.NewMap.initializeLayers(NewMap.java:714)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.v6.NewMap.onStart(NewMap.java:709)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1435)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at android.app.Activity.performStart(Activity.java:8222)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3818)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:246)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8506)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
02-28 23:18:51.730 17335 17335 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
02-28 23:18:51.730 17335 17335 W cgeo    : [main] UncaughtException
02-28 23:18:51.730 17335 17335 W cgeo    : java.lang.NullPointerException: Attempt to read from field 'android.net.Uri cgeo.geocaching.storage.ContentStorage$FileInformation.uri' on a null object reference
02-28 23:18:51.730 17335 17335 W cgeo    : 	at cgeo.geocaching.maps.mapsforge.v6.RenderThemeHelper.getThemeFor(RenderThemeHelper.java:183)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at cgeo.geocaching.maps.mapsforge.v6.RenderThemeHelper.reapplyMapTheme(RenderThemeHelper.java:134)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at cgeo.geocaching.maps.mapsforge.v6.NewMap.switchTileLayer(NewMap.java:642)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at cgeo.geocaching.maps.mapsforge.v6.NewMap.initializeLayers(NewMap.java:714)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at cgeo.geocaching.maps.mapsforge.v6.NewMap.onStart(NewMap.java:709)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1435)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at android.app.Activity.performStart(Activity.java:8222)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3818)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at android.os.Handler.dispatchMessage(Handler.java:106)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at android.os.Looper.loop(Looper.java:246)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at android.app.ActivityThread.main(ActivityThread.java:8506)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at java.lang.reflect.Method.invoke(Native Method)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
02-28 23:18:51.730 17335 17335 W cgeo    : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

### PR DESCRIPTION
**Pull request by eddiemuc**
fix #10093, rel to #10075: store mapthemes internal and fix npe on theme select

### CODE REVIEW

File: main/src/cgeo/geocaching/storage/LocalStorage.java [hunk 1]
```diff
@@ -449,6 +450,12 @@ protected Integer doInBackgroundInternal(final Void[] params) {
                 }
             }
 
+            //delete theme files from outdated internal theme file dir
```
Comment (Lineflyer) [inline main/src/cgeo/geocaching/storage/LocalStorage.java:453]: IMHO no need to implement a migration logic for nightly users only.
But might be good to be prepared if we need to make changes in future.

Comment (eddiemuc) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:453]: Really no migration? Some nightly users seem to care a lot about pollution of internal storage, there is no way to get outdated files out of it unless we do it in migration.

File: main/src/cgeo/geocaching/storage/LocalStorage.java [hunk 2]
```diff
@@ -187,7 +187,8 @@ public static File getOfflineLogImageDir(final String geocode) {
 
     @NonNull
     public static File getMapThemeInternalDir() {
-        final File dir = new File(getGeocacheDataDirectory("shared"), MAP_THEME_INTERNAL_DIR_NAME);
+        final File dir =  new File(getInternalCgeoDirectory(), MAP_THEME_INTERNAL_DIR_NAME);
+        //final File dir = new File(getGeocacheDataDirectory("shared"), MAP_THEME_INTERNAL_DIR_NAME);
```
[the remark this answers is not in the document]
Comment (eddiemuc) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:191]: yes, forgot that. will fix it.

[the remark this answers is not in the document]
Comment (eddiemuc) [reply main/src/cgeo/geocaching/storage/LocalStorage.java]: done

File: main/src/cgeo/geocaching/storage/LocalStorage.java (hunk 1 above)
Comment (Lineflyer) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:452]: mmh. The nightly user should be able to access `/Android/data/cgeo.geocaching/...`. But you might of course keep it in, I just wanted to make live easier as I think we only should really care about migrations between release versions as nightlies are always experimental.

Comment (eddiemuc) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:452]: Ok, maybe that was overkill. I can put it out again next time I am able to code, or we just merge it (it doesn't harm either way)

Comment (Lineflyer) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:452]: Lets keep it in...as you say. Does not harm.

**Pull request merged**